data/mods/ssb/abilities.js

```diff
@@ -815,6 +815,38 @@ let BattleAbilities = {
 			if (move.type === 'Flying') move.accuracy = true;
 		},
 	},
+	// Ransei
+	superguarda: {
+		desc: "This user's Attack is doubled until it is hit by a super effective attack. If this Pokemon is statused, its Attack is 1.5x; ignores burn halving physical damage. This Pokemon can only be damaged by direct attacks.",
+		shortDesc: "Atk 2x until hit by SE move. 1.5x Atk if statused. Immune to indirect dmg.",
+		id: "superguarda",
+		name: "Superguarda",
+		isNonstandard: "Custom",
+		onDamage(damage, target, source, effect) {
+			if (effect.effectType !== 'Move') {
+				return false;
+			}
+		},
+		onAfterDamage(damage, target, source, move) {
+			if (target.getMoveHitData(move).typeMod > 0) {
+				if (!target.heavilydamaged) {
```

Comment:
Pretty sure theres a `pokmeon.m` value we can make use of (used for OMs) to add values. I cant look into it now but will later if you are too busy.